pkg/cmd/monitor_command/run_monitor_command.go

```diff
@@ -95,7 +103,24 @@ func (opt *RunMonitorOptions) Run() error {
 	if err != nil {
 		return err
 	}
-	m, err := monitor.Start(ctx, restConfig, opt.AdditionalEventIntervalRecorders)
+	additionalEventIntervalRecorders := []monitor.StartEventIntervalRecorderFunc{
+		controlplane.StartAllAPIMonitoring,
+	}
+	if !opt.APIDisruptionOnly || os.Getenv("API_DISRUPTION_ONLY") != "true" {
+		additionalEventIntervalRecorders = append(additionalEventIntervalRecorders,
+			frontends.StartAllIngressMonitoring,
+			externalservice.StartExternalServiceMonitoring)
+	}
+	var lbTypeOpt string
+	if len(os.Getenv("LOAD_BALANCER_TYPE")) > 0 {
```

Comment:
no env vars.

Reply:
I couldn't find a better way for upgrades to pass.

Test container outside of the cluster runs latest version of the test code, so it has necessary CLI switches, but in the cluster we start with previous version of the payload. This means when in-cluster monitors are started we can't pass LB type / extra message via CLI args.

So the (terrible) solution is to use env vars, which don't make monitors crash when options are not available, until this change is backported to every version we could use.

Reply:
Not sure how to untangle this - maybe monitor should start in-cluster monitors itself and manupulate on their output without adding cli args / env vars?

Reply:
Created new subcommand - `run-disruption` with necessary cli args. In order to make it pass on upgrade tests the daemonset wraps it in bash with `|| true` to prevent failing "no crashlooping containers"